image/image.go

```diff
@@ -31,16 +32,22 @@ func ValidateLayout(src string, refs []string, out *log.Logger) error {
 	return validate(newPathWalker(src), refs, out)
 }
 
-// Validate walks through the given .tar file and validates the manifest
-// pointed to by the given refs or returns an error if the validation failed.
+// Validate opens the tar file given by the filename, then calls ValidateReader
 func Validate(tarFile string, refs []string, out *log.Logger) error {
 	f, err := os.Open(tarFile)
 	if err != nil {
 		return errors.Wrap(err, "unable to open file")
 	}
 	defer f.Close()
 
-	return validate(newTarWalker(tarFile, f), refs, out)
+	return ValidateReader(f, refs, out)
+}
+
+// ValidateReader walks through the stream and validates the manifest
+// pointed to by the given refs or returns an error if the validation failed.
+// The stream must be a valid .tar file.
+func ValidateReader(r io.ReadSeeker, refs []string, out *log.Logger) error {
```

Comment:
If we can't breakt the API, then I'd call this `ValidateReadSeeker` as it is more honest.